src/events/events.ts

```diff
@@ -0,0 +1,391 @@
+        await member.ban({
+            reason: ban
+        });
+    }
+};
+
+/**
+ * Logs member leave events to servers subscribed to logs.
+ * 
+ * @param client the Discord bot
+ * @param member the member who left
+ */
+export const onMemberLeave = async (client: Client, member: GuildMember): Promise<void> => {
+    // The server and logs channel ID of the server that the member left
+    const server: Server = await Server.get(member.guild.id);
+    const logsID: string = server.logs;
+
+    // Skip if the server isn't subscribed to logs
+    if (!logsID) {
+        return;
+    }
+
+    const logs: TextChannel = <TextChannel> client.channels.cache.get(logsID);
+
+    try {
+        // Log to the logs channel
+        await logs.send({ embeds: [genMemberLeaveEmbed(member)] });
+    } catch {
+        // Remove configurations if logging failed.
+        // Most likely causes are that the channel was deleted or that
+        // Sushi Bot had its permissions revoked
+        server.logs = null;
+        await server.save();
+    }
+};
+
+/**
+ * Logs member ban events to servers subscribed to logs.
+ * 
+ * @param client the Discord bot
+ * @param ban the ban data
+ */
+export const onMemberBan = async (client: Client, ban: GuildBan): Promise<void> => {
+    // The server and logs channel ID of the server that the member was banned from
+    const server: Server = await Server.get(ban.guild.id);
+    const logsID: string = server.logs;
+
+    // Skip if the server isn't subscribed to logs
+    if (!logsID) {
+        return;
+    }
+
+    const logs: TextChannel = <TextChannel> client.channels.cache.get(logsID);
+
+    try {
+        // Log to the logs channel
+        await logs.send({ embeds: [genMemberBanEmbed(ban)] });
+    } catch {
+        // Remove configurations if logging failed.
+        // Most likely causes are that the channel was deleted or that
+        // Sushi Bot had its permissions revoked
+        server.logs = null;
+        await server.save();
+    }
+};
+
+/**
+ * Logs member unban events to servers subscribed to logs.
+ * 
+ * @param client the Discord bot
+ * @param ban the ban data
+ */
+export const onMemberUnban = async (client: Client, ban: GuildBan): Promise<void> => {
+    // The server and logs channel ID of the server that the member was unbanned from
+    const server: Server = await Server.get(ban.guild.id);
+    const logsID: string = server.logs;
+
+    // Skip if the server isn't subscribed to logs
+    if (!logsID) {
+        return;
+    }
+
+    const logs: TextChannel = <TextChannel> client.channels.cache.get(logsID);
+
+    try {
+        // Log to the logs channel
+        await logs.send({ embeds: [genMemberUnbanEmbed(ban)] });
+    } catch {
+        // Remove configurations if logging failed.
+        // Most likely causes are that the channel was deleted or that
+        // Sushi Bot had its permissions revoked
+        server.logs = null;
+        await server.save();
+    }
+};
```

Comment:
These four listeners are quite similar. They get triggered when a server gains or loses a member and logs it to the moderation logs of servers subscribed to logs